### PR TITLE
[Fix Package] libirrlicht

### PR DESCRIPTION
**Pull request by saltedcoffii**

### CODE REVIEW

File: packages/irrlicht.rb [hunk 1]
```diff
@@ -13,23 +13,41 @@ class Irrlicht < Package
...
-      system 'make'
-      system 'make clean'
-      system 'make sharedlib'
+      system 'make sharedlib staticlib'
     end
   end
 
   def self.install
-    FileUtils.mkdir_p CREW_DEST_LIB_PREFIX
+    FileUtils.mkdir_p "#{CREW_DEST_LIB_PREFIX}"
```
Comment (uberhacker) [inline packages/irrlicht.rb:43]: This line doesn't need changed. :)

File: packages/irrlicht_examples.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  def self.patch
+    system "curl -#LO https://dev.gentoo.org/~mgorny/dist/irrlicht-1.8.4-patchset.tar.bz2"
+    @sha256sums = <<~EOF
+      03b6a5a8a98191f8efe6f7ccd5c957bbf2c4bf15ac6cdc987655792c8baff454  irrlicht-1.8.4-patchset.tar.bz2
+    EOF
+    IO.write("sha256sums", @sha256sums)
+    system "sha256sum -c sha256sums"
+    system "sed -i 's:\.\./\.\./media:../media:g' $(grep -rl '\.\./\.\./media' examples)"
+    system "sed -i 's/\r$//' source/Irrlicht/COSOperator.cpp"
+    system "dos2unix include/IrrCompileConfig.h"
```
Comment (uberhacker) [inline packages/irrlicht_examples.rb:23]: It's sad when so many contributors develop in Windoze. :)

File: packages/irrlicht_examples.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    system "tar xvf irrlicht-1.8.4-patchset.tar.bz2"
```
Comment (uberhacker) [inline packages/irrlicht_examples.rb:24]: Should we have a sha256sum check?

File: packages/irrlicht.rb (hunk 1 above)
Comment (saltedcoffii) [reply packages/irrlicht.rb:43]: It was an inconsistency that was annoying me. I also find it reduces my readability.

File: packages/irrlicht_examples.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    system "for patch in irrlicht-1.8.4-patchset/*; do patch -p 1 -i ${patch}; done"
+    system "sed -i 's:-I/usr/include/freetype2/:-I#{CREW_PREFIX}/include/freetype2:g' $(find -name Makefile)"
+    system "sed -i 's:-I/usr/X11R6/include:-I#{CREW_PREFIX}/X11R6/include:g' $(find -name Makefile)"
+    system "sed -i 's:-L/usr/X11R6/lib$(LIBSELECT):-L/usr/local/X11R6/lib$(LIBSELECT):g' $(find -name Makefile)"
+    system "sed -i 's:-I/usr/include:-I#{CREW_PREFIX}/include:g' $(find -name Makefile)"
```
Comment (uberhacker) [inline packages/irrlicht_examples.rb:29]: If you assigned `$(find -name Makefile)` to a variable and used that instead, would that speed the patch?

Comment (saltedcoffii) [reply packages/irrlicht_examples.rb:29]: I don't know, but the patch process seemed pretty quick anyway (a couple of seconds)

File: packages/irrlicht_tools.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    system "sha256sum -c sha256sums"
+    system "sed -i 's:\.\./\.\./media:../media:g' $(grep -rl '\.\./\.\./media' examples)"
+    system "sed -i 's/\r$//' source/Irrlicht/COSOperator.cpp"
+    system "dos2unix include/IrrCompileConfig.h"
+    system "tar xvf irrlicht-1.8.4-patchset.tar.bz2"
+    system "for patch in irrlicht-1.8.4-patchset/*; do patch -p 1 -i ${patch}; done"
+    system "sed -i 's:-I/usr/include/freetype2/:-I#{CREW_PREFIX}/include/freetype2:g' $(find -name Makefile)"
+    system "sed -i 's:-I/usr/X11R6/include:-I#{CREW_PREFIX}/X11R6/include:g' $(find -name Makefile)"
+    system "sed -i 's:-L/usr/X11R6/lib$(LIBSELECT):-L/usr/local/X11R6/lib$(LIBSELECT):g' $(find -name Makefile)"
+    system "sed -i 's:-I/usr/include:-I#{CREW_PREFIX}/include:g' $(find -name Makefile)"
```
Comment (uberhacker) [inline packages/irrlicht_tools.rb:29]: How about?
```
system "for m in \$(find -name Makefile); do \
    sed -i 's:-I/usr/include/freetype2/:-I#{CREW_PREFIX}/include/freetype2:g' \$m; \
    sed -i 's:-I/usr/X11R6/include:-I#{CREW_PREFIX}/X11R6/include:g' \$m; \
    sed -i 's:-L/usr/X11R6/lib$(LIBSELECT):-L/usr/local/X11R6/lib$(LIBSELECT):g' \$m; \
    sed -i 's:-I/usr/include:-I#{CREW_PREFIX}/include:g' \$m; done"
```

Comment (saltedcoffii) [reply packages/irrlicht_tools.rb]: Done.

File: packages/irrlicht.rb (hunk 1 above)
Comment (uberhacker) [reply packages/irrlicht.rb:44]: I thought we just made this change on a previous commit?

Comment (saltedcoffii) [reply packages/irrlicht.rb:44]: Would you like me to change it back?

Comment (uberhacker) [reply packages/irrlicht.rb:44]: No need but it would be less characters to wade through when checking.

**Pull request merged**